File: ios/TriPay/AdvancedAnalytics.swift
import Foundation

struct AdvancedAnalytics {
    // Calculate Z-Score for anomaly detection
    static func detectAnomalies(transactions: [Transaction]) -> [Transaction] {
        let amounts = transactions.map { $0.amount }
        let count = Double(amounts.count)
        guard count > 1 else { return [] }
        
        let mean = amounts.reduce(0, +) / count
        let sumOfSquaredDiffs = amounts.map { pow($0 - mean, 2) }.reduce(0, +)
        let standardDeviation = sqrt(sumOfSquaredDiffs / count)
        
        return transactions.filter { tx in
            let zScore = abs(tx.amount - mean) / standardDeviation
            return zScore > 2.0 // Simple threshold for spikes
        }
    }
    
    // Simple Linear Trend for end-of-month prediction
    static func predictMonthlySpend(transactions: [Transaction], totalDaysInMonth: Int) -> Double {
        let currentSpend = transactions.reduce(0) { $0 + $1.amount }
        let dates = transactions.map { $0.date }
        guard let firstDate = dates.min(), let lastDate = dates.max() else { return currentSpend }
        
        let daysElapsed = Calendar.current.dateComponents([.day], from: firstDate, to: lastDate).day ?? 1
        let dailyRate = currentSpend / Double(max(1, daysElapsed))
        
        return dailyRate * Double(totalDaysInMonth)
    }
}

File: android/app/src/main/kotlin/com/tripay/logic/AdvancedAnalytics.kt
package com.tripay.logic

import com.tripay.models.Transaction
import kotlin.math.pow
import kotlin.math.sqrt

object AdvancedAnalytics {
    fun detectAnomalies(transactions: List<Transaction>): List<Transaction> {
        val amounts = transactions.map { it.amount }
        val count = amounts.size
        if (count <= 1) return emptyList()

        val mean = amounts.average()
        val stdDev = sqrt(amounts.map { (it - mean).pow(2.0) }.sum() / count)

        return transactions.filter { tx ->
            val zScore = Math.abs(tx.amount - mean) / stdDev
            zScore > 2.0
        }
    }
}
